callout/src/test/java/com/google/apigee/JavaCalloutRemotePolicyExecutionTest.java

```diff
@@ -0,0 +1,227 @@
+import static org.mockito.Mockito.doReturn;
+
+import com.apigee.flow.execution.Action;
+import com.apigee.flow.execution.ExecutionContext;
+import com.apigee.flow.execution.ExecutionResult;
+import com.apigee.flow.message.Message;
+import com.apigee.flow.message.MessageContext;
+import com.google.apigee.Execute.Execution;
+import com.google.protobuf.TextFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class JavaCalloutRemotePolicyExecutionTest {
+  private static final String KEY = "Example";
+  private static final String VALUE = "Value";
+  private static final String URL_PROPERTY_NAME = "remote_execution_url";
+  private static final String SERVER_URL = "url";
+  @Mock private MessageContext messageContext;
+  @Mock private ExecutionContext executionContext;
+  @Mock private RemotePolicyExecutionHandler remotePolicyExecutionHandler;
+  @Mock private Message message;
+  private JavaCalloutRemotePolicyExecution callout;
+
+  @Before
+  public void init() {
+    MockitoAnnotations.openMocks(this);
+
+    Map<String, String> properties = new HashMap<>();
+    properties.put(URL_PROPERTY_NAME, SERVER_URL);
+    callout = new JavaCalloutRemotePolicyExecution(properties, remotePolicyExecutionHandler);
+  }
+
+  @Test
+  public void testExecution() throws Exception {
+    Execution.Builder executionBuilder = Execution.newBuilder();
+    TextFormat.merge(
+        "messageContext {"
+            + "  target_request_message {"
+            + "    flow_variables {"
+            + "      key: \"" + KEY + "\""
+            + "      value {"
+            + "        flow_variable: \"" + VALUE + "\""
+            + "      }"
+            + "    }"
+            + "  }"
+            + "}"
+            + "executionContext {"
+            + "}"
+            + "executionResult {"
+            + "    action: CONTINUE"
+            + "}",
+        executionBuilder);
+    Execution execution = executionBuilder.build();
+
+    Execution.Builder mockExecutionBuilder = Execution.newBuilder();
+    TextFormat.merge(
+        "messageContext {\n"
+            + "}\n"
+            + "executionContext {\n"
+            + "}", mockExecutionBuilder);
+    Execution mockExecution = mockExecutionBuilder.build();
+
+    ExecutionResult expected = new ExecutionResult(true, Action.CONTINUE);
+    expected.setErrorResponse("");
+    expected.setErrorResponseHeaders(new HashMap<>());
+    expected.setProperties(new Properties());
+
+    doReturn(execution)
+        .when(remotePolicyExecutionHandler)
+        .sendRemoteHttpServerRequest(mockExecution, SERVER_URL);
+    doReturn(message).when(messageContext).getMessage();
+    doNothing().when(message).setContent(VALUE);
+
+    ExecutionResult actual = callout.execute(messageContext, executionContext);
+
+    assertTrue(isSameExecutionResult(expected, actual));
```

Comment:
Can you include some verify statements to verify that the setContent method was called in the MessageContext mock?

Reply:
We should do this with the rest of the functions as well.

Reply:
Actually, if the ExecutionResult isn't Continue, we don't call setContent at all. I've removed the doNothing's from those tests.